### DISCUSSION TITLE
Klipper install suggestions

### DISCUSSION THREAD

**Discussion by ClearAirTurbulence3D** [bassamanator/Sovol-SV06-firmware#2] [General]
I'm in the process of installing Klipper. The instructions have been good, so far. I would move or expand the RPi setup right after (or even before) the printer firmware install. I'm not using git, so I'm doing it all manually.

I'm using MainsailOS, so I'm following the RPi setup [here](https://docs.mainsail.xyz/setup/mainsailos/pi-imager) and the Klipper [install](https://docs.mainsail.xyz/setup/mainsailos/klipper-setup).

When Mainsail boots, it'll have many errors, since it won't have any printer info I ran the update first, then copied the printer.cfg from here, then created the cfgs folder and uploaded all the files (drag and drop works for files but not folders in the Mainsail web interface).

**Comment by ClearAirTurbulence3D** [comment]
One issue with the Klipper install is that some older RPis will assign a different USB name to the virtual serial port; the correct name has to be determined with ls /dev/serial/by-id/* and the correct port name entered in printer,cfg

**Reply by wendyr21** [reply]
for whatever reason its not listed for me when i do that. I am running mutliple instances. any ideas?

**Reply by bassamanator** [reply]
> for whatever reason its not listed for me when i do that. I am running mutliple instances. any ideas?
Are you saying that when you do a ` ls /dev/serial/by-id/`, nothing is found?

Klipper must be flashed to the board for that command to show results.

**Reply by ClearAirTurbulence3D** [reply]
Just to clarify, Bassam's Klipper firmware is in the misc folder (klipper.bin). Copy it to a flash card, rename it (I used kilpper1.bin, but it's not critical), make sure it's the only .bin file on the card. Then put it in the SV06, power up and wait a few seconds (30 - 60 should be enough)

**Comment by bassamanator** [comment]
Missed this post. Thanks for posting! I have the instructions on how to find the correct port listing in the `printer.cfg`, but they should be on the main instruction page, will list them there as well. Thanks again.

**Reply by bassamanator** [reply]
Added these instructions to README in commit [4e4cc47](https://github.com/bassamanator/Sovol-SV06-firmware/commit/4e4cc4794a3677b5e6ccd7d0b5af8f2673ac6fac).

**Comment by wendyr21** [comment]
thanks :) 
    On Sunday, March 5, 2023 at 10:10:51 PM EST, ClearAirTurbulence3D ***@***.***> wrote:  
 
 


Just to clarify, Bassam's Klipper firmware is in the misc folder (klipper.bin). Copy it to a flash card, rename it (I used kilpper1.bin, but it's not critical), make sure it's the only .bin file on the card. Then put it in the SV06, power up and wait a few seconds (30 - 60 should be enough)
